#### Backward generation(BWD)

1. generate data after integral

In [1]:
import time

import numpy as np

from backward import generate_bwd

### Testing ...

In [2]:
!wc -l ./data/bwd_sample5000.txt

    1994 ./data/bwd_sample5000.txt


In [10]:
from multiprocessing.pool import ThreadPool

_FINISH = False

sequences_per_process = 64
process_runs = 8

print('generating {} samples in total'.format(
    sequences_per_process * process_runs)
)

start = time.time()
with ThreadPool(processes=14) as p:
    out = p.map(generate_bwd, [sequences_per_process]*process_runs)
    time.sleep(1)
    print('done')
    _FINISH = True
    p.terminate()
    p.join()

length = time.time()-start
print("Finished: ", length)  

seconds_per_seq = length / (sequences_per_process * process_runs)
print(seconds_per_seq * 1000000 / 3600)

generating 512 samples in total
process finished
process finished
process finished
process finished
process finished
process finished
process finished
process finished
done
Finished:  100.57037425041199
54.5629200577322


In [11]:
len(out)

8

<pre>
processes=14
sequences_per_process   process_runs         hours for 1 million 
- 16                       4                 94.28
- 4                        16                54.06
- 128                      4                 80.07
- 32                       16                58.18
- 64                       8                 54.56
</pre>

In [ ]:
file = open(r"./data/bwd_sample2048.txt", "w+")
for i in range(len(out)):
    sequence =out[i]
    file.writelines(sequence) 

In [ ]:
filenames = []
for i in range(1):
    filename = "./data/bwd_sample1028"  + ".txt"
    filenames.append(filename)
with open(r"./data/bwd_sample5000.txt", 'w+') as outfile:
    for fname in filenames:
        with open(fname) as infile:
            outfile.write(infile.read())

In [ ]:
!wc -l ./data/bwd_sample5000.txt